include/dfu/flash_img.h

```diff
@@ -24,6 +24,18 @@ struct flash_img_context {
 #endif
 };
 
+/**
+ * @brief Get a flash sector which comprises a given offset.
+ *
+ * @param fap the flash area where the offset belongs
+ * @param off the offset
+ * @param sector the buffer to store the retrieved sector info
+ *
+ * @return 0 on success, negative errno code on fail.
+ */
+int flash_sector_from_off(struct flash_area const *fap, off_t off,
```

Comment:
pleas take https://github.com/zephyrproject-rtos/zephyr/pull/20973/ into account.
offset will be in relation to the fap beginning.